rocAL/source/file_io_stream.cpp

```diff
@@ -0,0 +1,61 @@
+
+#include <errno.h>
+#include <sys/stat.h>
+#include <cstdio>
+#include <cstring>
+#include <memory>
+#include <string>
+#include "pipeline/exception.h"
+#include "file_io_stream.h"
+#include "pipeline/commons.h"
+#include <iostream>
+
+std::unique_ptr<FileIOStream> FileIOStream::open(const std::string& path) {
+    return std::make_unique<FileIOStream>(path);
+}
+
+FileIOStream::FileIOStream(const std::string& path) : _path(path) {
```

Comment:
Why are we not using std::filesystem from c++17 for this? It should give you all the standard operations for reading to buffer, size, writing to buffer. Please explore this option: https://en.cppreference.com/w/cpp/filesystem

Reply:
Done